pkg/apis/volumesnapshot/v1beta1/types.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package v1beta1
 
 import (
 	core_v1 "k8s.io/api/core/v1"
```

Comment:
```Go
	// Source represents the location and type of the volume snapshot
	VolumeSnapshotSource `json:",inline" protobuf:"bytes,1,opt,name=volumeSnapshotSource"`		VolumeSnapshotSource `json:",inline" protobuf:"bytes,1,opt,name=volumeSnapshotSource"`
...
	// PersistentVolumeRef represents the PersistentVolume that the snapshot has been		// PersistentVolumeRef represents the PersistentVolume that the snapshot has been
	// taken from. It becomes non-nil when VolumeSnapshot and VolumeSnapshotContent are bound.		// taken from. It becomes non-nil when VolumeSnapshot and VolumeSnapshotContent are bound.
	// +optional		// +optional
	PersistentVolumeRef *core_v1.ObjectReference `json:"persistentVolumeRef" protobuf:"bytes,3,opt,name=persistentVolumeRef"`		PersistentVolumeRef *core_v1.ObjectReference `json:"persistentVolumeRef" protobuf:"bytes,3,opt,name=persistentVolumeRef"`
```

`VolumeSnapshotSource` allows for more than one type of source. `PersistentVolumeRef` only allows PV type. Should be consistent. Or just remove `PersistentVolumeRef`.

Reply:
P0: unless there are use cases

Reply:
use case: volume dependent snapshots

Reply:
Can you explain this in more detail?

Reply:
In one of the regular snapshot community meetings, two engineers brought up the following use case:
both storage class reclaim policy and snapshot deletion policy are retain, i.e., delete PVC/VolumeSnapshot will not delete corresponding PV/SnapshotContent
1. create VolumeSnapshot on an existing PVC
2. delete PVC and VolumeSnapshot, PV & SnapshotContent remain in the system
3. storage admin or whoever is privileged decided to delete PV, and they will fail in cases where snapshot is tightly coupled with the volume
    in underlying storage system. (those storage systems requires all snapshot to be deleted before a volume could be deleted). ATM, CSI driver
    would return error with a message like "Snapshot exists, volume could not be deleted" but will not return the list of existing snapshot ids on the 
    underlying storage system.
 now storage admin would need to know which snapshots to delete, and without the volume ref, it would be tricky to identify which
 SnapshotContents to delete, before they could actually clean up PV.